File: hyoungkyu/230720/[BOJ] 1198 삼각형으로 자르기.py
# 실버2 / 52ms
import sys
input = sys.stdin.readline

N = int(input())
dots = [list(map(int, input().split())) for _ in range(N)]
i = 0

# 넓이 구하는 공식
def func(lst):
    lst += [lst[0]]
    tmp = 0
    for i in range(3):
        tmp += lst[i][0]*lst[i+1][1] - lst[i+1][0]*lst[i][1]
    return abs(tmp)/2

maxV = 0
for i in range(N-2):
    for j in range(i+1, N-1):
        for k in range(j+1, N):
            lst = [dots[i], dots[j], dots[k]]
            tmp = func(lst)
            maxV = max(maxV, tmp)
print(maxV)

File: hyoungkyu/230720/[BOJ] 1198 삼각형으로 자르기.java
// 실버2 / 128ms
import java.io.*;

class Main
{
	public static void main(String args[]) throws Exception
	{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        int[][] dots = new int[N][2];

        for (int i=0; i<N; i++) {
            String[] line = br.readLine().split(" ");
            dots[i][0] = Integer.parseInt(line[0]);
            dots[i][1] = Integer.parseInt(line[1]);
        }

        double maxV = 0;
        for (int i=0; i<N-2; i++) {
            for (int j=i+1; j<N-1; j++) {
                for (int k=j+1; k<N; k++) {
                    int[][] lst = {dots[i], dots[j], dots[k], dots[i]};
                    double tmp = func(lst);
                    if (maxV < tmp) maxV = tmp;
                }
            }
        }
        System.out.println(maxV);
	}

    public static double func(int[][] lst) {
        int tmp = 0;
        for (int i=0; i<3; i++) {
            tmp += (lst[i][0] * lst[i + 1][1] - lst[i + 1][0] * lst[i][1]);
        }
        return Math.abs(tmp) * 1.0 / 2;
    }
}